src/middlewared/middlewared/plugins/api_key.py

```diff
@@ -183,12 +163,12 @@ async def _validate(self, schema_name, data, id_=None):
     def _generate(self):
         return "".join([random.SystemRandom().choice(string.ascii_letters + string.digits) for _ in range(64)])
 
-    def _serve(self, data, key):
+    def _serve(self, data: ApiKeyEntry, key: str | None) -> ApiKeyEntry:
         if key is None:
             return data
 
         return dict(data, key=f"{data['id']}-{key}")
```

Comment:
Is `data` a `dict` or an `ApiKeyEntry`?

Reply:
I would propose sticking to dictionaries at the current stage and only changing API declarations. Thoughts, @yocalebo ?

No need to revert already made changes in any case; however, let's make type hints reflect the reality. If the function accepts a dict, let it be a dict.

Reply:
Sounds good, I will fix this.